pkgs/tools/security/jd-gui/default.nix

```diff
@@ -1,56 +1,49 @@
-{ stdenv, fetchurl, gradle_2_5, perl, makeWrapper, jre, makeDesktopItem, writeText, runtimeShell }:
+{ stdenv, fetchFromGitHub, jre, jdk, makeDesktopItem, writeText, runtimeShell }:
 
 let
-  version = "1.4.0";
-  name = "jd-gui-${version}";
-
-  src = fetchurl {
-    url    = "https://github.com/java-decompiler/jd-gui/archive/v${version}.tar.gz";
-    sha256 = "0anz7szlr5kgmsmkyv34jdynsnk8v6kvibcyz98jsd96fh725lax";
+  pname = "jd-gui";
+  version = "1.6.5";
+
+  src = fetchFromGitHub {
+    owner = "java-decompiler";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0yn2xcwznig941pw2f3wi8ixz1wprxcn9wl0g2ggdzx51rfwgzzi";
   };
 
-  deps = stdenv.mkDerivation {
-    name = "${name}-deps";
+  jar = stdenv.mkDerivation rec {
+    name = "${pname}-jar";
     inherit src;
-    nativeBuildInputs = [ gradle_2_5 perl ];
 
-    buildPhase = ''
-      export GRADLE_USER_HOME=$(mktemp -d);
-      gradle --no-daemon build
-    '';
-
-    # Mavenize dependency paths
-    # e.g. org.codehaus.groovy/groovy/2.4.0/{hash}/groovy-2.4.0.jar -> org/codehaus/groovy/groovy/2.4.0/groovy-2.4.0.jar
-    installPhase = ''
-      find $GRADLE_USER_HOME/caches/modules-2 -type f -regex '.*\.\(jar\|pom\)' \
-        | perl -pe 's#(.*/([^/]+)/([^/]+)/([^/]+)/[0-9a-f]{30,40}/([^/\s]+))$# ($x = $2) =~ tr|\.|/|; "install -Dm444 $1 \$out/$x/$3/$4/$5" #e' \
-        | sh
-    '';
+    nativeBuildInputs = [ jdk ];
 
-    outputHashAlgo = "sha256";
-    outputHashMode = "recursive";
-    outputHash = "1apmqiphnav79m4rdii58h7f4qslpfig4qybyyl2fr7zk92gv3l9";
-  };
+    patchPhase = "patchShebangs gradlew";
 
-  # Point to our local deps repo
-  gradleInit = writeText "init.gradle" ''
-    logger.lifecycle 'Replacing Maven repositories with ${deps}...'
+    gradleInit = writeText "init.gradle" ''
+      logger.lifecycle 'Add flags for reproducible archives'
 
-    gradle.projectsLoaded {
-      rootProject.allprojects {
-        buildscript {
-          repositories {
-            clear()
-            maven { url '${deps}' }
+      gradle.projectsLoaded {
+        rootProject.allprojects {
+          tasks.withType(AbstractArchiveTask) {
+            preserveFileTimestamps = false
+            reproducibleFileOrder = true
           }
         }
-        repositories {
-          clear()
-          maven { url '${deps}' }
-        }
       }
-    }
-  '';
+    '';
+
+    buildPhase = ''
+      export JAVA_HOME=${jdk}
+      export GRADLE_USER_HOME=.
+      ./gradlew --no-daemon --init-script ${gradleInit} jar
+    '';
+
+    installPhase = "cp build/libs/${pname}-${version}.jar $out";
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "flat";
+    outputHash = "16wh7cmkhi305i3x3l8qkmcy9z9l67q3rqaxdsy96gih9gbmmzb5";
```

Comment:
This seems a bit flacky. Does it not break the checksum, when a new javac changes the way the byte code is compiled?

Reply:
Yes, unfortunately. Everything is back to the old approach.